cdap-master/src/main/java/co/cask/cdap/data/tools/UpgraderMain.java

```diff
@@ -340,9 +347,22 @@ private void addModules(DatasetFramework datasetFramework) throws DatasetManagem
   }
 
   /**
-   * Creates the {@link Constants#DEFAULT_NAMESPACE} namespace
+   * Creates the {@link Constants#SYSTEM_NAMESPACE} in hbase and {@link Constants#DEFAULT_NAMESPACE} namespace and also
+   * adds it to the store
    */
-  private void createDefaultNamespace() {
+  private void createNamespaces() {
```

Comment:
Is there a chance that some of the upgrade functions (especially those related to HBase tables) are performed before this method is called?